go.mod

```diff
@@ -777,7 +776,6 @@ require (
 	go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp v0.44.0 // indirect
 	go.opentelemetry.io/otel/exporters/stdout/stdoutmetric v0.44.0 // indirect
 	go.opentelemetry.io/otel/exporters/stdout/stdouttrace v1.21.0 // indirect
-	go4.org/unsafe/assume-no-moving-gc v0.0.0-20231121144256-b99613f794b6 // indirect
```

Comment:
This is important to have. See https://github.com/go4org/intern/blob/3eb7198706b254cddea46d066da7316b2620ba7a/intern.go#L141. The internal string interner uses the same trickery, and without this we would not know if these assumptions are broken in a new go release.

Of course we can fix this separately, but I don't think this PR is the place to do it.

Reply:
reverted